src/Microsoft.DotNet.Cli.Utils/CommandResolution/MSBuildProject.cs

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Evaluation;
+using NuGet.Frameworks;
+using NuGet.ProjectModel;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    internal class MSBuildProject : IProject
+    {
+        private Project _project;
+
+        private string _msBuildExePath;
+
+        public string DepsJsonPath
+        {
+            get
+            {
+                return _project
+                    .AllEvaluatedProperties
+                    .FirstOrDefault(p => p.Name.Equals("_ProjectDepsFilePath"))
+                    .EvaluatedValue;
+            }
+        }
+
+        public string RuntimeConfigJsonPath
+        {
+            get
+            {
+                return _project
+                    .AllEvaluatedProperties
+                    .FirstOrDefault(p => p.Name.Equals("_ProjectRuntimeConfigFilePath"))
+                    .EvaluatedValue;
+            }
+        }
+
+        public Dictionary<string, string> EnvironmentVariables
+        {
+            get
+            {
+                return new Dictionary<string, string>
+                {
+                    { Constants.MSBUILD_EXE_PATH, _msBuildExePath }
+                };
+            }
+        }
+
+        public MSBuildProject(
+            string msBuildProjectPath,
+            NuGetFramework framework,
+            string configuration,
+            string outputPath,
+            string msBuildExePath)
+        {
+            var globalProperties = new Dictionary<string, string>()
+            {
+               { "MSBuildExtensionsPath", Path.GetDirectoryName(msBuildExePath) }
+            };
+
+            if(framework != null)
+            {
+                globalProperties.Add("TargetFramework", framework.GetShortFolderName());
+            }
+
+            if(outputPath != null)
+            {
+                globalProperties.Add("OutputPath", outputPath);
+            }
+
+            if(configuration != null)
+            {
+                globalProperties.Add("Configuration", configuration);
+            }
+
+            _project = ProjectCollection.GlobalProjectCollection.LoadedProjects.FirstOrDefault(p =>
```

Comment:
So, without this, I was running into issues when doing new Project because this had been called before for the same project, which caused new Project to throw an exception.